internal/providers/pke/pkeworkflow/update_node_group_activity.go

```diff
@@ -0,0 +1,277 @@
+func (a UpdateNodeGroupActivity) Execute(ctx context.Context, input UpdateNodeGroupActivityInput) (UpdateNodeGroupActivityOutput, error) {
+	if input.NodePoolName == "master" {
+		return UpdateNodeGroupActivityOutput{}, errors.New("updating master node pool is not yet supported")
+	}
+
+	providerSecret, err := brn.Parse(input.SecretID)
+	if err != nil {
+		return UpdateNodeGroupActivityOutput{}, errors.WrapIf(err, "failed to parse secret BRN")
+	}
+	sess, err := a.sessionFactory.New(providerSecret.OrganizationID, providerSecret.ResourceID, input.Region)
+	if err = errors.WrapIf(err, "failed to create AWS session"); err != nil { // internal error?
+		return UpdateNodeGroupActivityOutput{}, err
+	}
+
+	cluster, err := a.clusters.GetCluster(ctx, input.ClusterID)
+	if err != nil {
+		return UpdateNodeGroupActivityOutput{}, err
+	}
+
+	awsCluster, ok := cluster.(AWSCluster)
+	if !ok {
+		return UpdateNodeGroupActivityOutput{}, errors.New(fmt.Sprintf("can't get AWS client for %t", cluster))
+	}
+
+	_, signedToken, err := a.tokenGenerator.GenerateClusterToken(input.OrganizationID, input.ClusterID)
+	if err != nil {
+		return UpdateNodeGroupActivityOutput{}, errors.WrapIf(err, "can't generate Pipeline token")
+	}
+
+	nodeLabels := []string{
+		fmt.Sprintf("%v=%v", pkgCluster.NodePoolNameLabelKey, input.NodePoolName),
+	}
+
+	if input.NodePoolVersion != "" {
+		nodeLabels = append(nodeLabels, fmt.Sprintf("%v=%v", pkgCluster.NodePoolVersionLabelKey, input.NodePoolVersion))
+	}
+
+	bootstrapCommand, err := awsCluster.GetBootstrapCommand(
+		input.NodePoolName,
+		a.externalBaseURL,
+		a.externalBaseURLInsecure,
+		signedToken,
+		nodeLabels,
+		input.Version,
+	)
+	if err != nil {
+		return UpdateNodeGroupActivityOutput{}, errors.WrapIf(err, "failed to fetch bootstrap command")
+	}
+
+	template, err := cloudformation2.GetCloudFormationTemplate(PKECloudFormationTemplateBasePath, WorkerCloudFormationTemplate)
+	if err != nil {
+		return UpdateNodeGroupActivityOutput{}, errors.WrapIf(err, "loading CF template")
+	}
+
+	cloudformationClient := cloudformation.New(sess)
+
+	stackParams := []*cloudformation.Parameter{
+		{
+			ParameterKey:     aws.String("KeyName"),
+			UsePreviousValue: aws.Bool(true),
+		},
+		sdkCloudFormation.NewOptionalStackParameter(
+			"ImageId",
+			input.NodeImage != "",
+			input.NodeImage,
+		),
+		{
+			ParameterKey:     aws.String("InstanceType"),
+			UsePreviousValue: aws.Bool(true),
+		},
+		{
+			ParameterKey:     aws.String("NodeSpotPrice"),
+			UsePreviousValue: aws.Bool(true),
+		},
+		{
+			ParameterKey:     aws.String("MinSize"),
+			UsePreviousValue: aws.Bool(true),
+		},
+		{
+			ParameterKey:     aws.String("MaxSize"),
+			UsePreviousValue: aws.Bool(true),
+		},
+		// sdkCloudFormation.NewOptionalStackParameter(
+		// 	"NodeAutoScalingGroupMaxBatchSize",
+		// 	input.MaxBatchSize > 0,
+		// 	fmt.Sprintf("%d", input.MaxBatchSize),
+		// ),
```

Comment:
**Question**: why is this commented out? (Do we need it? Does it have any value as a comment?)

Reply:
Yes, just removed it, once (if ever) we will have an open-source version of the upgrade flow we will probably used it (leftover from EKS).